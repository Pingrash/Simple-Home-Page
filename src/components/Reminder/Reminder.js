import React from 'react';
import PropTypes from 'prop-types';

import Paper from '@material-ui/core/Paper';
import Typography from '@material-ui/core/Typography';
import Button from '@material-ui/core/Button';
import { makeStyles } from '@material-ui/styles';

const useStyles = makeStyles(theme => ({
  container: {
    display: 'flex',
    flexDirection: 'row',
    padding: '1rem',
    marginTop: '1rem'
  },
  title: {
    flex: 1,
    textAlign: 'left',
    margin: 'auto 1rem',
    textTransform: 'capitalize'
  },
  body: {
    flex: 4,
    textAlign: 'left',
    margin: 'auto 0'
  },
  button: {
    flex: 1
  }
}));

const Reminder = ({ title, body, deleteReminder }) => {
  const classes = useStyles();
  return (
    <Paper className={classes.container}>
      <Typography
        variant='h6'
        className={classes.title}
        data-test='title'
      >
        {title}
      </Typography>
      <Typography
        variant='body1'
        className={classes.body}
        data-test='body'
      >
        {body}
      </Typography>
      <Button
        color='secondary'
        onClick={() => deleteReminder(title)}
        className={classes.button}
        data-test='button'
      >
        Delete
      </Button>
    </Paper>
  );
};

Reminder.propTypes = {
  title: PropTypes.string.isRequired,
  body: PropTypes.string.isRequired,
  deleteReminder: PropTypes.func.isRequired
};

export default Reminder;
