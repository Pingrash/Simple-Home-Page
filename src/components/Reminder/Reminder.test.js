import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import Reminder from './Reminder';
import {
  findByDataTest,
  checkProps,
  testFunc
} from '../../../Utilities';

const props = {
  title: 'title',
  body: 'body',
  deleteReminder: testFunc
};

const setup = () => {
  const component = shallow(<Reminder {...props} />);
  return component;
};

describe('Snapshot', () => {
  it('Should match snapshot', () => {
    const tree = renderer.create(<Reminder {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('Reminder renders correctly', () => {
  let wrapper;
  beforeEach(() => (wrapper = setup()));

  it('Should render without errors', () => {
    expect(wrapper.length).toBe(1);
  });

  it('Should render a title', () => {
    const title = findByDataTest(wrapper, 'title');
    expect(title.length).toBe(1);
  });

  it('Should render a body', () => {
    const body = findByDataTest(wrapper, 'body');
    expect(body.length).toBe(1);
  });

  it('Should render a button', () => {
    const btn = findByDataTest(wrapper, 'button');
    expect(btn.length).toBe(1);
  });
});

describe('Reminder has correct props', () => {
  it('Should not return a props error', () => {
    const propsError = checkProps(Reminder, props);
    expect(propsError).toBeUndefined();
  });
});
