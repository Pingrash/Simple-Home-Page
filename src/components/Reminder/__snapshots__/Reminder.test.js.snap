// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot Should match snapshot 1`] = `
<div
  className="MuiPaper-root makeStyles-container-1 MuiPaper-elevation1 MuiPaper-rounded"
>
  <h6
    className="MuiTypography-root makeStyles-title-2 MuiTypography-h6"
    data-test="title"
  >
    title
  </h6>
  <p
    className="MuiTypography-root makeStyles-body-3 MuiTypography-body1"
    data-test="body"
  >
    body
  </p>
  <button
    className="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-button-4 MuiButton-textSecondary"
    data-test="button"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <span
      className="MuiButton-label"
    >
      Delete
    </span>
  </button>
</div>
`;
